plugin/tags/tags.go

```diff
@@ -27,21 +28,29 @@ func New() *tags {
 // tags is the actual tags plugin that maintains a map with all
 // tags from all processed pages.
 type tags struct {
-	m map[string]*model.ListPage
+	tags      map[string]*model.ListPage
+	tagsMutex sync.RWMutex
 }
 
 // ProcessPage creates a new map entry for each tag in the processed
 // page and adds the page to the entry's list page.
 func (t *tags) ProcessPage(page *model.Page) error {
 	for _, tag := range page.Tags {
-		//sanitizing the tags like "Making Coffee" to "making-coffee"
+		// Sanitizing the tags like "Making Coffee" to "making-coffee".
 		tag.Name = strings.Replace(tag.Name, " ", "-", -1)
 		tag.Name = strings.ToLower(tag.Name)
 
-		if _, exists := t.m[tag.Name]; !exists {
+		t.tagsMutex.RLock()
+		_, tagExists := t.tags[tag.Name]
+		t.tagsMutex.RUnlock()
+
+		if !tagExists {
 			t.createListPage(tag.Name)
 		}
-		t.m[tag.Name].Pages = append(t.m[tag.Name].Pages, page)
+
+		t.tagsMutex.Lock()
```

Comment:
Based on the assumption from the above comment, these lines become

```go
t.tagsMutext.RUnlock()
t.tagsMutex.Lock()
```

-- so we may remove this unlock-lock-sequence in the middle entirely.

Reply:
Yes, good point!